capstone/src/main/webapp/script.js

```diff
@@ -110,25 +130,211 @@ async function getAnalysis(text, lang, requestedAttributes, tokenizer) {
+  const toxicityOfOriginal = await getOriginalToxicity(text, lang, analysisContainer, loadingEl);
+  if (!toxicityOfOriginal) {
+    return;
+  }
+  
+  // Get Datamuse API replacements of substring
+  const replacements = await callDatamuse(substringForReplacements, wordType);
+  if (replacements.length === 0) {
+    analysisContainer.removeChild(loadingEl)
+    analysisContainer.appendChild(createAnyElement('b', 'Datamuse API did not find any words or phrases matching your query'));
+    return;
+  }
+ 
+  printReplacements(text, substringForReplacements, replacements, toxicityOfOriginal, lang, analysisContainer, loadingEl);
+}
+
+/** Prints the modified strings to the page with their Perspective TOXICITY scorings */
+async function printReplacements(text, substringForReplacements, replacements, toxicityOfOriginal, lang, container, loadingEl) {
+  // Get Perspective scores on new sentences
+  const promises = [];
+  const newSentences = [];
+  for (let i = 0; i < replacements.length; i++) {
+    const replacement = replacements[i].word;
+    const newSentence = text.replace(substringForReplacements, replacement);
+    newSentences.push(newSentence);
+    promises.push(callPerspective(newSentence, lang, ['TOXICITY']));
+  }
+
+  await Promise.all(promises).then(resolvedPromises => {
+    // Calculate and sort the score differences of the new sentences
+    const sentenceData = [];
+    for (let i = 0; i < resolvedPromises.length; i++) {
+      const toxicityOfReplacement = resolvedPromises[i].attributeScores.TOXICITY.summaryScore.value;
+      const scoreDiff = toxicityOfReplacement - toxicityOfOriginal;
+      sentenceData.push({
+        replacement: replacements[i].word,
+        sentence: newSentences[i], 
+        scoreDiff: scoreDiff,
+      });
+    }
+    sentenceData.sort(function(a, b) {
+      return a.scoreDiff - b.scoreDiff;
+    });
+ 
+    container.removeChild(loadingEl);
+
+    // Print the new sentences and their score differences
+    for (let i = 0; i < sentenceData.length; i++) {
+      printSentence(container, sentenceData[i])
+    }
+
+    // Draw the chart
+    drawDatamuseChart(resolvedPromises, substringForReplacements, toxicityOfOriginal, replacements);
+  });
+}
+ 
+/** Prints a sentence replacement with the correct style and data  */ 
+function printSentence(container, sentenceData) {
+  const isPositive = sentenceData.scoreDiff > 0;
+  const className = isPositive ? 'red-background' : 'green-background';
+  const sign = isPositive ? '+' : '';
+  const styledSentence = sentenceData.sentence.replace(sentenceData.replacement, '<b>' + sentenceData.replacement + '</b>');
+  const scoreEl = '<span class="' + className + '">' +  sign + decimalToPercentage(sentenceData.scoreDiff) + '</span>';
+  container.appendChild(createAnyElement('p', styledSentence + ' -> ' + scoreEl));
+}
+
+/** Gives the user extra options after they get the replacement data */
+function setUpExtras(text, lang, substringForReplacements) {
```

Comment:
Nit, I think this function could use a more descriptive name, maybe `showDataMuseWordReplacementOptions`?

Reply:
Switched it to that